zeebe/msgpack-value/src/test/java/io/camunda/zeebe/msgpack/BasePropertyTest.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Camunda License 1.0. You may not use this file
+ * except in compliance with the Camunda License 1.0.
+ */
+package io.camunda.zeebe.msgpack;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import io.camunda.zeebe.msgpack.property.BaseProperty;
+import io.camunda.zeebe.msgpack.value.StringValue;
+import java.nio.charset.StandardCharsets;
+import org.junit.jupiter.api.Test;
+
+final class BasePropertyTest {
+  @Test
+  void shouldMaskSanitizedValuesInToString() {
+    // given
+    final TestProperty property = new TestProperty("non-sanitized-value").sanitized();
+
+    // when
+    final String result = property.toString();
+
+    // then
+    assertThat(property.isSanitized()).isTrue();
+    assertThat(result).isEqualTo("test-key => ***");
+  }
+
+  @Test
+  void shouldNotMaskNonSanitizedValuesInToString() {
+    // given
+    final TestProperty property = new TestProperty("non-sanitized-value");
+
+    // when
+    final String result = property.toString();
+
+    // then
+    assertThat(property.isSanitized()).isFalse();
+    assertThat(result).isEqualTo("test-key => non-sanitized-value");
```

Comment:
🔧 Can we also have some tests for the writeJson with true vs false?

Reply:
In my defense, there were 0 tests before :grin: